subprojects/internal-testing/src/main/groovy/org/gradle/testing/internal/util/RetryRule.groovy

```diff
@@ -54,6 +54,9 @@ class RetryRule implements MethodRule {
 
     @Override
     Statement apply(Statement base, FrameworkMethod method, Object target) {
+        if (specification == null) {
+            specification = target as Specification
```

Comment:
Yes. I will check it and do that if it does not cause any harm. I only added it originally, because I missed that we can use the `target` parameter of `apply` here.